bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,317 @@
+import functools
+import logging
+import random
+import string
+from asyncio import TimeoutError
+from datetime import datetime
+from pathlib import Path
+from types import TracebackType
+from typing import Callable, Dict, List, Optional, Set, Tuple, Type
```

Comment:
Instead of importing all of these one by one, we can do `import typing as t`.
```py
def __exit__(
        self,
        exc_type: t.Optional[t.Type[BaseException]],
        exc_value: t.Optional[BaseException],
        traceback: t.Optional[TracebackType]
    ):
        if len(self.cache[self.word]) == 0:
            logging.info(f"No rhymes were found for the word: {self.word}"

Reply:
Is this absolutely necessary? I prefer specifying the types in the imports so that I don't have to add `t.` in the code.

Reply:
No it is not, but yeah it is a "Could do".